pkg/controller/dynamicupgradeableconditionscontroller.go

```diff
@@ -0,0 +1,300 @@
+	if err != nil {
+		log.Printf("Error listing cluster extensions: %v", err)
+		return
+	}
+
+	// Get all ClusterExtensions incompatible with next Y-stream
+	for _, obj := range ceList {
+		ce, ok := obj.(*ocv1alpha1.ClusterExtension)
+		if !ok {
+			log.Printf("Error: could not cast to ClusterExtension object")
+			return
+		}
+		store := buildHelmStore(ce, c.kubeclient.CoreV1().Secrets(ce.Namespace))
+
+		rel, _ := store.Deployed(ce.Name)
+		props, err := propertyListFromPropertiesAnnotation(rel.Chart.Metadata.Annotations["olm.properties"])
+		if err != nil {
+			log.Printf("Error converting olm.properties: %v", err)
+			return
+		}
+		for _, p := range props {
+			if p.Type == MaxOpenShiftVersionProperty {
+				version := string(p.Value)
+				maxOCPVersion, err := toSemver(version)
+				if err != nil {
+					log.Printf("Error converting to semver for version %s: %v", version, err)
+					continue
+				}
+				if maxOCPVersion == nil || maxOCPVersion.GTE(next) {
+					// All good
+				} else {
+					// Incompatible
+					name := rel.Labels[BundleNameKey]
+					incompatibleOperators = append(incompatibleOperators, name)
+				}
+			}
+		}
+	}
+
+	updateStatusFuncs := make([]v1helpers.UpdateStatusFunc, 0, 1)
+	if len(incompatibleOperators) > 0 {
+		updateStatusFuncs = append(updateStatusFuncs, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    "Upgradeable",
+			Status:  operatorv1.ConditionFalse,
+			Reason:  incompatibleOperatorsInstalled,
+			Message: strings.Join(incompatibleOperators, ","),
+		}))
+
+		if _, _, updateErr := v1helpers.UpdateStatus(context.TODO(), c.operatorClient, updateStatusFuncs...); updateErr != nil {
+			log.Printf("Error listing secrets: %v", err)
+			return
+		}
+	} else {
+		updateStatusFuncs = append(updateStatusFuncs, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:   "Upgradeable",
+			Status: operatorv1.ConditionTrue,
+		}))
+
+		if _, _, updateErr := v1helpers.UpdateStatus(context.TODO(), c.operatorClient, updateStatusFuncs...); updateErr != nil {
+			log.Printf("Error listing secrets: %v", err)
+			return
+		}
+	}
+	return
+}
+
+func propertyListFromPropertiesAnnotation(raw string) ([]property.Property, error) {
+	var props []property.Property
+	if err := json.Unmarshal([]byte(raw), &props); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal properties annotation: %w", err)
+	}
+	return props, nil
+}
+
+func buildHelmStore(ce *ocv1alpha1.ClusterExtension, secretClient v1.SecretInterface) helm.Storage {
+	csConfig := storage.ChunkedSecretsConfig{
+		ChunkSize:      0,
+		MaxReadChunks:  0,
+		MaxWriteChunks: 0,
+		Log:            nil,
+	}
+
+	owner := ce.Name
+	ownerRefs := []metav1.OwnerReference{*metav1.NewControllerRef(ce, ce.GetObjectKind().GroupVersionKind())}
+	ownerRefSecretClient := helmclient.NewOwnerRefSecretClient(secretClient, ownerRefs, func(secret *corev1.Secret) bool {
+		return secret.Type == storage.SecretTypeChunkedIndex
+	})
```

Comment:
I don't think you need this part. If I remember correctly, this is only used to write release secrets. We're only reading from storage, so none of this ends up being used.

Reply:
I don't think there's an interface where I can omit this. In order for the helm.Storage to be defined I need a driver defined which requires all this jazz